### ISSUE TITLE
Benchmark results on imagenet validation set

### ISSUE THREAD

**Issue by fengfu-chris** [lukemelas/EfficientNet-PyTorch#12]
Here is my benchmark results on imagenet validation set:

model | inference time | image load time | bs | #batches | Memory(MB) | top-1 | top-5 | top-1(paper) | top-5 (paper) | notes |
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
EfficientNet-B0 | 17.1 | 16.4 | 1 | 50000 | 913 | 75.48 | 92.48 | 76.3 | 93.2 | resize shorter size to 224，keep aspect ratio
EfficientNet-B3 | 26.5 | 16.8 | 1 | 50000 | 1403 | 80.47 | 95.16 | 81.1 | 95.5 | resize shorter size to 224, keep aspect ratio
EfficientNet-B0 | 86.2 | 954.3 | 64 | 782 | 2325 | 75.81 | 92.66 | 76.3 | 93.2 | resize to 224+32, keep aspect ratio, then center crop to 224x224
EfficientNet-B3 | 304.2 | 1070.8 | 64 | 782 | 4913 | 80.28 | 94.98 | 81.1 | 95.5 | resize to 300+44, keep aspect ratio, then center crop to 300x300

time unit: ms / batch

A little bit worse than original benchmarks (B3 can reach around 80.9% top-1).

**Comment by lukemelas** [comment]
Thanks for the benchmark results! Would it be possible for you to include your computing setup alongside the results? 

Also, I want to make a note that there's an open PyTorch issue that may be relevant -- grouped convs on FP32 are slow, and will be improved in the near future: pytorch/pytorch#18631

**Comment by fengfu-chris** [comment]
@lukemelas Sure，Below are the settings and results.

Test Environment:  
OS: Centos 7.6  
CUDA: 9.0  
cuDNN: 7.0  
PyTorch: 1.1.0  
TorchVision: 0.2.0  
GPU: GTX 1080Ti  

Setting 1 :   
Read images one by one to fit a batch, then do forward pass with batch of images. (like examples/simple/example.py)

The results are listed above. 

Setting 2:    
Using torchvision.datasets.ImageFolder to read images sequentially (like examples/imagenet/main.py). Number of workers is 4 (default). BS=64, #batches=782 (i.e. 50000 images total). The results are below.


Model | Total Time(s) | Memory(MB) | crop_padding size |Resize Interpolation | top-1 | top-5 | top-1(paper) | top-5(paper)
-- | -- | -- | -- | -- | -- | -- | -- | -- 
EfficientNet-B0 | 213.2 | 2331 | 0 | BICUBIC | 75.88 | 92.72 | 76.3 | 93.2 
EfficientNet-B0 | - | - | 32 | BICUBIC | 76.53 | 93.01 | - | -
EfficientNet-B1 | 206.5 | 2619 | 0 | BICUBIC | 78.20 | 93.90 | 78.8 | 94.4 
EfficientNet-B1 | - | - | 32 | BICUBIC | 78.55 | 94.10 | - | -
EfficientNet-B2 | 225.2 | 2997 | 0 | BICUBIC | 79.34 | 94.65 | 79.8 | 94.9
EfficientNet-B2 | - | - | 32 | BICUBIC | 79.61 | 94.71 | - | -
EfficientNet-B3 | 250.8 | 5017 | 0 | BICUBIC | 80.78 | 95.25 | 81.1 | 95.5
EfficientNet-B3 | - | - | 32 | BICUBIC | 80.97 | 95.27 | - | -

NOTE:  
1. the resize interpolation function is very IMPORTATNT. In seettings1, the interpolation is BILINEAR in PyTorch by default. While in the original TF EfficientNet repo., the interpolation function is BICUBIC.
2. the crop_padding size also plays IMPORTANT role for good performance.
3. While theEfficientNet-B3 torch model has 0.2 top-1 gap compared to the result in the paper, we find that the original tensorflow model also has the performance drop (around 80.9).

**Comment by skyw** [comment]
> @lukemelas Sure，below are the settings and results.
> 
> Testing Environment:
> OS: Centos 7.6
> CUDA: 9.0
> cuDNN: 7.0
> PyTorch: 1.1.0
> TorchVision: 0.2.0
> GPU: GTX 1080Ti
> 
> Setting 1 :
> Read images one by one to fit a batch, then do forward pass with batch of images. (refer to examples/simple/example.py)
> 
> The results are listed above.
> 
> Setting 2:
> Using torchvision.datasets.ImageFolder to read images sequentially (refer to examples/imagenet/main.py). Number of workers is 4 (default). BS=64, #batches=782 (i.e. 50000 images total). The results are below.
> 
> Model	Total Time(s)	Memory(MB)	crop_padding size	Resize Interpolation	top-1	top-5	top-1(paper)	top-5(paper)
> EfficientNet-B0	213.2	2331	0	BICUBIC	75.88	92.72	76.3	93.2
> EfficientNet-B0	-	-	32	BICUBIC	76.53	93.01	-	-
> EfficientNet-B1	206.5	2619	0	BICUBIC	78.20	93.90	78.8	94.4
> EfficientNet-B1	-	-	32	BICUBIC	78.55	94.10	-	-
> EfficientNet-B2	225.2	2997	0	BICUBIC	79.34	94.65	79.8	94.9
> EfficientNet-B2	-	-	32	BICUBIC	79.61	94.71	-	-
> EfficientNet-B3	250.8	5017	0	BICUBIC	80.78	95.25	81.1	95.5
> EfficientNet-B3	-	-	32	BICUBIC	80.97	95.27	-	-
> NOTE:
> 
> 1. The resize interpolation function is very IMPORTATNT. In seettings1, the interpolation is BILINEAR in PyTorch by default. While in the original TF EfficientNet repo., the interpolation function is BICUBIC.
> 2. The crop_padding size also plays IMPORTANT role for good performance.
> 3. While the EfficientNet-B3 torch model has 0.2 top-1 gap compared to the result in the paper, we find that the original tensorflow model also has the performance drop (around 80.9).
> 4. For EfficientNet-B0, the speed bottleneck lies in reading images.

HI, could you also provide your preprocess code? I can't reproduce the result with padding. Central crop is used for validation, while padding doesn't affect central crop?

**Comment by skyw** [comment]
It seems to work if I increase resize size by 32.
